OpenSim/Common/DataTable.h

```diff
@@ -121,6 +121,163 @@ class DataTable_ : public AbstractDataTable {
         *this = std::move(*table);
     }
 
+    /** Contruct DataTable_<double, double> from 
+    DataTable_<double, ThatETY> where ThatETY can be SimTK::Vec<X>. Each column
+    of the other table is split into multiple columns of this table. For example
+    , DataTable_<double, Vec3> with 3 columns and 4 rows will construct
+    DataTable<double, double> of 9 columns and 4 rows where each component of
+    SimTK::Vec3 ends up in one column. Column labels of the resulting DataTable
+    will use column labels of source table appended with suffixes provided.
+
+    \tparam ThatETY Datatype of the matrix underlying the given DataTable.
+
+    \param that DataTable to copy-construct this table from. This table can be
+                of for example DataTable_<double, Quaternion>, 
+                DataTable_<double, Vec6>.
+    \param suffixes Suffixes to be used for column-labels in this table when
+                    splitting columns of 'that' table. For example a column 
+                    labelled 'marker' from DataTable_<double, Vec3> will be
+                    split into 3 columns named
+                    \code
+                    std::string{'marker' + suffixes[0]},
+                    std::string{'marker' + suffixes[1]},
+                    std::string{'marker' + suffixes[2]}
+                    \endcode
+
```

Comment:
What happens to the meta data associated with *that* table? Other than labels, I think any associated meta data should be preserved. If that is already the case a brief comment to that effect would be good.